lib/matplotlib/widgets.py

```diff
@@ -1967,15 +1967,12 @@ def set_animated(self, val):
 
     def closest(self, x, y):
         """Return index and pixel distance to closest index."""
-        pts = np.transpose((self.x, self.y))
+        pts = np.column_stack([self.x, self.y])
         # Transform data coordinates to pixel coordinates.
         pts = self.ax.transData.transform(pts)
-        diff = pts - ((x, y))
-        if diff.ndim == 2:
-            dist = np.sqrt(np.sum(diff ** 2, axis=1))
-            return np.argmin(dist), np.min(dist)
-        else:
```

Comment:
I don't quite understand this else branch. How could `diff.ndim != 2` happen?

Reply:
Probably that was intended to handle the case some time ago where self.x and self.y were scalars rather than arrays, but now even in that case we get a 2d result anyways.
Well, the branch gets deleted in the rewrite :)